README.rst
===================
cmsplugin_yandexmap
===================

cmsplugin_yandexmap - это плагин для django-cms. Позволяет добавлять на страницу интерактивную Яндекс карту. 
Возможности аналогичны плагину googlemap входящему в состав данной CMS.

Лицензия MIT.

Зависимости
===========

* django-cms >= 3.0

Установка
=========

Как обычно::

    $ pip install cmsplugin_yandexmap

или ::

    $ easy_install cmsplugin_yandexmap

или ::

    $ hg clone http://github.org/moskrc/cmsplugin_yandexmap/
    $ cd cmsplugin_yandexmap
    $ python setup.py install


Добавить плагин в ``INSTALLED_APPS``::

    INSTALLED_APPS = (
        ...
        'cmsplugin_yandexmap',
        ...
    )

Запустить ``syncdb``, или ``migrate`` если вы используете South.

HISTORY
Changelog for cmsplugin_yandexmap
=====================================

1.0.0 (2011-03-06)
----------------

* Initial project setup


1.3.1 (2014-12-20)
----------------

* django-cms 3.0
* yandex map api 2.1
* migrating to github

cmsplugin_yandexmap/models.py
from django.db import models
from django.utils.translation import ugettext_lazy as _
from cms.models import CMSPlugin

class YandexMap(CMSPlugin):
    """
    A yandex maps integration
    """
    title = models.CharField(_("map title"), max_length=100, blank=True, null=True)
    
    address = models.CharField(_("address"), max_length=150)
    zipcode = models.CharField(_("zip code"), max_length=30)
    city = models.CharField(_("city"), max_length=100)
    
    content = models.CharField(_("additional content"), max_length=255, blank=True, null=True)
    zoom = models.IntegerField(_("zoom level"), blank=True, null=True)
    
    lat = models.DecimalField(_('latitude'), max_digits=10, decimal_places=6, null=True, blank=True, 
        help_text=_('Use latitude & longitude to fine tune the map possiton.'))
    lng = models.DecimalField(_('longitude'), max_digits=10, decimal_places=6, null=True, blank=True)
    
    
    def __unicode__(self):
        return u"%s (%s, %s %s)" % (self.get_title(), self.address, self.zipcode, self.city,)
    
    def get_title(self):
        if self.title == None:
            return _("Map")
        return self.title
    
    def get_content(self):
        if self.content == None:
            return ""
        return self.content
    
    def get_zoom_level(self):
        if self.zoom == None:
            return 13
        return self.zoom
    
    def get_lat_lng(self):
        if self.lat and self.lng:
            return [self.lat, self.lng]

cmsplugin_yandexmap/cms_plugins.py
from django.conf import settings
from cms.plugin_pool import plugin_pool
from cms.plugin_base import CMSPluginBase
from django.utils.translation import ugettext_lazy as _
from models import YandexMap
from django.forms.widgets import Media


class YandexMapPlugin(CMSPluginBase):
    model = YandexMap
    name = _("Yandex Map")
    render_template = "cmsplugin_yandexmap/yandexmap.html"
    
    def render(self, context, instance, placeholder):
        context.update({
            'object': instance, 
            'placeholder': placeholder, 
            'settings': settings,
        })
        return context
    
    #def get_plugin_media(self, request, context, plugin):
    #    return Media(js = ('http://maps.google.com/maps/api/js?sensor=true',))

plugin_pool.register_plugin(YandexMapPlugin)


cmsplugin_yandexmap/migrations/0001_initial.py
# encoding: utf-8
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models

class Migration(SchemaMigration):

    def forwards(self, orm):
        
        # Adding model 'YandexMap'
        db.create_table('cmsplugin_yandexmap', (
            ('cmsplugin_ptr', self.gf('django.db.models.fields.related.OneToOneField')(to=orm['cms.CMSPlugin'], unique=True, primary_key=True)),
            ('title', self.gf('django.db.models.fields.CharField')(max_length=100, null=True, blank=True)),
            ('address', self.gf('django.db.models.fields.CharField')(max_length=150)),
            ('zipcode', self.gf('django.db.models.fields.CharField')(max_length=30)),
            ('city', self.gf('django.db.models.fields.CharField')(max_length=100)),
            ('content', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True)),
            ('zoom', self.gf('django.db.models.fields.IntegerField')(null=True, blank=True)),
            ('lat', self.gf('django.db.models.fields.DecimalField')(null=True, max_digits=10, decimal_places=6, blank=True)),
            ('lng', self.gf('django.db.models.fields.DecimalField')(null=True, max_digits=10, decimal_places=6, blank=True)),
        ))
        db.send_create_signal('cmsplugin_yandexmap', ['YandexMap'])


    def backwards(self, orm):
        
        # Deleting model 'YandexMap'
        db.delete_table('cmsplugin_yandexmap')


    models = {
        'cms.cmsplugin': {
            'Meta': {'object_name': 'CMSPlugin'},
            'creation_date': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'language': ('django.db.models.fields.CharField', [], {'max_length': '15', 'db_index': 'True'}),
            'level': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'lft': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'parent': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.CMSPlugin']", 'null': 'True', 'blank': 'True'}),
            'placeholder': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['cms.Placeholder']", 'null': 'True'}),
            'plugin_type': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'}),
            'position': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True', 'blank': 'True'}),
            'rght': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'}),
            'tree_id': ('django.db.models.fields.PositiveIntegerField', [], {'db_index': 'True'})
        },
        'cms.placeholder': {
            'Meta': {'object_name': 'Placeholder'},
            'default_width': ('django.db.models.fields.PositiveSmallIntegerField', [], {'null': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'slot': ('django.db.models.fields.CharField', [], {'max_length': '50', 'db_index': 'True'})
        },
        'cmsplugin_yandexmap.yandexmap': {
            'Meta': {'object_name': 'YandexMap', 'db_table': "'cmsplugin_yandexmap'", '_ormbases': ['cms.CMSPlugin']},
            'address': ('django.db.models.fields.CharField', [], {'max_length': '150'}),
            'city': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'cmsplugin_ptr': ('django.db.models.fields.related.OneToOneField', [], {'to': "orm['cms.CMSPlugin']", 'unique': 'True', 'primary_key': 'True'}),
            'content': ('django.db.models.fields.CharField', [], {'max_length': '255', 'null': 'True', 'blank': 'True'}),
            'lat': ('django.db.models.fields.DecimalField', [], {'null': 'True', 'max_digits': '10', 'decimal_places': '6', 'blank': 'True'}),
            'lng': ('django.db.models.fields.DecimalField', [], {'null': 'True', 'max_digits': '10', 'decimal_places': '6', 'blank': 'True'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '100', 'null': 'True', 'blank': 'True'}),
            'zipcode': ('django.db.models.fields.CharField', [], {'max_length': '30'}),
            'zoom': ('django.db.models.fields.IntegerField', [], {'null': 'True', 'blank': 'True'})
        }
    }

    complete_apps = ['cmsplugin_yandexmap']
